### PR TITLE
adds the option to disable task logs

### PR DESCRIPTION
**Pull request by AugustoPeres**

### CODE REVIEW

File: inductiva/tasks/streams.py [hunk 1]
```diff
@@ -109,6 +111,10 @@ def __on_message(self, ws: websocket.WebSocketApp, message: str):
                 return
             self._write_message(msg)
 
+    def _disable_logs(self):
+        tasks_api.TasksApi(api.get_client()).disable_task_logs(
+            {"task_id": self.task_id})
+
```
Comment (sergiomsantos) [inline inductiva/tasks/streams.py:117]:
```suggestion
    def _disable_logs(self):
        client = api.get_client()
        payload = {"task_id": self.task_id}
        tasks_api.TasksApi(client).disable_task_logs(payload)

```

**Review by sergiomsantos — CHANGES_REQUESTED** [review]

File: inductiva/tasks/streams.py [hunk 2]
```diff
@@ -268,6 +275,9 @@ def _run_websocket(self):
         and update the start query to get the logs from the last
         received timestamp."""
 
+        for sig in [signal.SIGTERM, signal.SIGINT, signal.SIGHUP]:
+            signal.signal(sig, self._handle_termination)
```
Comment (Pbarbosa92) [inline inductiva/tasks/streams.py:279]: Why don't we just do 
```
if self._end_of_stream or self._keyboard_interrupt:
      self._disable_logs()
      break
```
I don't think there is a way to capture a sigkill.
@sergiomsantos can give a better input on this.

File: inductiva/tasks/streams.py [hunk 3]
```diff
@@ -109,6 +112,16 @@ def __on_message(self, ws: websocket.WebSocketApp, message: str):
                 return
             self._write_message(msg)
 
+    def _handle_termination(self, signum, _):
+        print(f"Received termination signal {signum}.")
+        self._disable_logs()
+        sys.exit()
```
Comment (sergiomsantos) [inline inductiva/tasks/streams.py:118]: let's remove this method.

File: inductiva/tasks/task.py [hunk 4]
```diff
@@ -161,6 +161,10 @@ def get_info(self) -> Dict[str, Any]:
 
         return info
 
+    def disable_logs(self) -> None:
+        """Disables logs of the task."""
+        self._api.disable_task_logs({"task_id": self.id})
```
Comment (sergiomsantos) [inline inductiva/tasks/task.py:166]: not used.

**Review by sergiomsantos — CHANGES_REQUESTED** [review]
let's keep the initial version. If needed, we can iterate afterwards.

**Review by sergiomsantos — APPROVED** [review]

**Pull request closed without merging**